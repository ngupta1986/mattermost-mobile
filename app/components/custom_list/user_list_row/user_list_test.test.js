// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserListRow from './user_list_row';

configure({adapter: new Adapter()});

jest.mock('react-intl');
jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

jest.mock('rn-fetch-blob', () => ({
    fs: {
        dirs: {
            DocumentDir: () => jest.fn(),
            CacheDir: () => jest.fn(),
        },
    },
}));

jest.mock('rn-fetch-blob/fs', () => ({
    dirs: {
        DocumentDir: () => jest.fn(),
        CacheDir: () => jest.fn(),
    },
}));

describe('UserListRow', () => {
    const formatMessage = jest.fn();
    const baseProps = {
        id: '123455',
        isMyUser: false,
        user: {
            id: '21345',
            username: 'user',
            delete_at: 0,
        },
        theme: {},
        teammateNameDisplay: 'test',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <UserListRow {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot for deactivated user', () => {
        const deactivatedUser = {
            id: '21345',
            username: 'user',
            delete_at: 100,
        };

        const newProps = {
            ...baseProps,
            user: deactivatedUser,
        };

        const wrapper = shallow(
            <UserListRow {...newProps}/>,
            {context: {intl: {formatMessage}}},
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot for  currentUser with (you) populated in list', () => {
        const newProps = {
            ...baseProps,
            isMyUser: true,
        };

        const wrapper = shallow(
            <UserListRow {...newProps}/>,
            {context: {intl: {formatMessage}}},
        );
        expect(wrapper).toMatchSnapshot();
    });
});
