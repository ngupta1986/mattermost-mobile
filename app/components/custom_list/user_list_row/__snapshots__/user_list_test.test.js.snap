// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListRow should match snapshot 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <CustomListRow
    enabled={true}
    id="21345"
    onPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#3d3c40",
          "flexDirection": "row",
          "marginLeft": 10,
        }
      }
    >
      <Connect(ProfilePicture)
        size={32}
        userId="21345"
      />
    </Component>
    <Component
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component>
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 15,
            }
          }
        >
          @user
        </Component>
      </Component>
    </Component>
    <Component
      style={
        Object {
          "width": 25,
        }
      }
    />
  </CustomListRow>
</Component>
`;

exports[`UserListRow should match snapshot for  currentUser with (you) populated in list 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <CustomListRow
    enabled={true}
    id="21345"
    onPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#3d3c40",
          "flexDirection": "row",
          "marginLeft": 10,
        }
      }
    >
      <Connect(ProfilePicture)
        size={32}
        userId="21345"
      />
    </Component>
    <Component
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component>
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 15,
            }
          }
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "width": 25,
        }
      }
    />
  </CustomListRow>
</Component>
`;

exports[`UserListRow should match snapshot for deactivated user 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <CustomListRow
    enabled={true}
    id="21345"
    onPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#3d3c40",
          "flexDirection": "row",
          "marginLeft": 10,
        }
      }
    >
      <Connect(ProfilePicture)
        size={32}
        userId="21345"
      />
    </Component>
    <Component
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component>
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 15,
            }
          }
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "width": 25,
        }
      }
    />
  </CustomListRow>
</Component>
`;
